File: url-shortener-back/src/main/java/br/com/urlshortener/domain/model/UserURL.java
package br.com.urlshortener.domain.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "user_url")
public class UserURL {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id", unique = true)
	private Long id;

	@Column(name = "original_url")
	private String originalUrl;

	@Column(name = "generated_url")
	private String generatedUrl;

	@Column(name = "date")
	@Temporal(TemporalType.TIMESTAMP)
	private Date date;

	@Column(name = "code")
	private String code;

	@Column(name = "user_id")
	@ManyToOne
	private User user;

}

File: url-shortener-frontend/src/app/model/user-url.ts
import { User } from "./user";

export class UserURL {

    id: number;
    originalUrl: string;
    generatedUrl: string;
    date: Date;
    code: string;
    user: User;

    constructor(id?: number, originalUrl?: string, generatedUrl?: string,
        date?: Date, code?: string, user?: User) {
        this.id = id;
        this.originalUrl = originalUrl;
        this.generatedUrl = generatedUrl;
        this.date = date;
        this.code = code;
        this.user = user;
    }

}